### ISSUE TITLE
Set Map Layers dynamically 

### ISSUE THREAD

**Issue by maxammann** [digitalfabrik/entitlementcard#750]
**Is your feature request related to a problem? Please describe.**
The clusters on the  map are right now bavaria blue. We might want to set the particular theme color. Currently the layers are hardcoded in the `styles.json` and should set dynamically to set the cluster color depending on the projects build configuration (theme color)

**Describe the solution you'd like**
- create dynamic layers in flutter (as we do in react-native)
- set the particular theme color for the cluster layer
- set the icon directly for each poi by just setting the category in the symbol layer. We may consider categoryId and projectId
- ensure legacy support (maintain styles.json)

**Describe alternatives you've considered**
None

**Additional context**
Current hardcoded layer solution styles.json (https://github.com/digitalfabrik/ehrenamtskarte-maplibre-style/blob/22cc66f0a9a361372496fe2f2d29d4fb0ee70fdf/style.json#L1597)
Here are some infos how to do it
https://github.com/maplibre/flutter-maplibre-gl/blob/6eb27181f2ecfc34a883c74ef681e61f8e3f2fb3/example/lib/layer.dart#L228
This would also enable to customize the cluster colors per project.
Additionally we could just pass the categoryId or favoriteIcon as icon-image as we do for integreat

**Comment by sarahsporck** [comment]
Is there currently a way to pass arguments to the map styles from within the app? Or would this be done by using env variables?

**Comment by maxammann** [comment]
Hm, I think there are already different styles for the map. One for entitlementcard and one for ehrenamtskarte. If not we can do that I think.

They get deployed and we just have to change the url in the build config.

**Comment by f1sh1918** [comment]
hm not sure how this works here in flutter but you maybe could also provide these cluster layers in the app instead of the style.json as we do in integreat @sarahsporck . That would it make easy to pass the theme colors we want. I think to have fore each new whitelabel project a separate map-libre style repo looks seems very complicated

**Comment by sarahsporck** [comment]
I agree with you.And this should be fairly easy by just adding some functioanality to the MapStyleHandler.
